tests/test_runs/test_run.py

```diff
@@ -179,3 +183,31 @@ def test_to_from_filesystem_no_model(self):
         # assert default behaviour is throwing an error
         with self.assertRaises(ValueError, msg='Could not find model.pkl'):
             openml.runs.OpenMLRun.from_filesystem(cache_path)
+
+    def test_publish_with_local_loaded_flow(self):
+        """
+        Publish a run tied to a local flow after it has first been saved to
+         and loaded from disk.
+        """
+        model = Pipeline([
+            ('imputer', Imputer(strategy='mean')),
+            ('classifier', DummyClassifier()),
+        ])
+        task = openml.tasks.get_task(119)
+        run = openml.runs.run_model_on_task(
+            model=model,
+            task=task,
+            add_local_measures=False,
+            avoid_duplicate_runs=False,
+            upload_flow=False
+        )
+
+        cache_path = os.path.join(
+            self.workdir,
+            'runs',
+            str(random.getrandbits(128)),
+        )
+        run.to_filesystem(cache_path)
+        # obtain run from filesystem
+        loaded_run = openml.runs.OpenMLRun.from_filesystem(cache_path)
+        loaded_run.publish()
```

Comment:
And that it is instead uploaded here? Maybe you could add some checks on the run, too? Maybe using the mock package, then you do not even need to interact with the server.

Reply:
(although it might be good to test this kind of behavior against the server)

Reply:
Well this is where I am a little puzzled by the desire to both test against the server but also use mocking instead 😅  Anyway, I updated the test right now to ensure the run and flow exist on the server.

Reply:
Yes, most likely it is good the way it is